cpp/src/plasma/external_store.cc

```diff
@@ -0,0 +1,64 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <iostream>
+#include <sstream>
+
+#include "external_store.h"
+
+namespace plasma {
+
+std::string ExternalStores::ExtractStoreName(const std::string& endpoint) {
+  size_t off = endpoint.find_first_of(':');
+  if (off == std::string::npos) {
+    throw std::invalid_argument("Malformed endpoint " + endpoint);
+  }
+  return endpoint.substr(0, off);
+}
+
+void ExternalStores::RegisterStore(const std::string& store_name,
+                                   std::shared_ptr<ExternalStore> store) {
+  std::cerr << "Registering external store \"" << store_name << "\"" << std::endl;
```

Comment:
Let's replace this with ARROW_LOG(INFO)

Reply:
This code is called during static initialization, when ARROW_LOG isn’t initialized yet. I can remove the log message altogether.

Reply:
Ah ok makes sense. In that case I'd remove these two messages. The possible providers are known at compile time anyways and not modified at run time, right?

Reply:
No, they are not; I'll remove the messages.